src/libraries/System.Runtime/tests/System.Runtime.Extensions.Tests/System/Convert.FromHexString.cs

```diff
@@ -136,6 +136,18 @@ public static void TooShortDestination()
             Assert.Equal(destinationSize, bytesWritten);
         }
 
+        [Fact]
+        public static void TooLongDestination()
+        {
+            string hex = Convert.ToHexString([255, 255, 255]);
+            byte[] buffer = new byte[100];
+            var status = Convert.FromHexString(hex, buffer, out int charsConsumed, out int bytesWritten);
+
+            Assert.Equal(OperationStatus.NeedMoreData, status);
```

Comment:
I believe that this assertion is incorrect. `NeedMoreData` should only be returned if the parser handled partial data (i.e. there's an odd number of hex chars). If the buffer fits the result it should always return `Done`.